src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrScannerTest.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Darc.Tests.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.DotNet.DarcLib.Models.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Darc.Tests.VirtualMonoRepo;
+
+[TestFixture]
+public class VmrScannerTest : VmrTestsBase
+{
+    [Test]
+    public async Task VmrScannerFindsNothingTest()
+    {
+        await InitializeRepoAtLastCommit(Constants.ProductRepoName, ProductRepoPath);
+
+        var list = await CallDarcScan();
+        Assert.AreEqual(0, list.Count);
+    }
+
+    [Test]
+    public async Task VmrScannerVmrPreservedTest()
+    {
+        await InitializeRepoAtLastCommit(Constants.ProductRepoName, ProductRepoPath);
+
+        var newFilePath = VmrPath / "src" / Constants.ProductRepoName / "src";
+        Directory.CreateDirectory(newFilePath);
+        File.WriteAllText(newFilePath / "test.dll", "this is a test file");
+        File.WriteAllText(newFilePath / ".gitattributes", $"*.dll {VmrInfo.KeepAttribute}");
+        await GitOperations.CommitAll(VmrPath, "Commit dll file");
+
+        var list = await CallDarcScan();
+        Assert.AreEqual(0, list.Count);
```

Comment:
Please use `FluentAssertions` and also use assertions for collections directly (e.g. `list.Should().BeEmpty()`)